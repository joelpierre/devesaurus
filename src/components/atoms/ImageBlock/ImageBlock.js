import React from 'react';
import PropTypes from 'prop-types';

const ImageBlock = (props) => {
  return (<div data-test="component-image-block"/>);
};

ImageBlock.defaultProps = {};

ImageBlock.propTypes = {};

export default ImageBlock;
