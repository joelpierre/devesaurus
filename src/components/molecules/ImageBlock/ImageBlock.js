import React from 'react';

function ImageBlock({ module, pageTheme }) {
  if (module.__typename !== 'image_block') {
    return null;
  }

  return (
    <section className={`primary-main__section theme--${module.theme} image-block`}>
      <div className="container-fluid">
        <div className="row">
          <div className="flex text-center">
            <h2>Image Block</h2>
          </div>
        </div>
      </div>
    </section>
  );
}

export default ImageBlock;
