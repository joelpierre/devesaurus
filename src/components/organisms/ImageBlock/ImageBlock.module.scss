@import "~sass/core";

.image-block {
  $root: &;

  position: relative;
  width: 100%;
}
