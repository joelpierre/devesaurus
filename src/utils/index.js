export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
