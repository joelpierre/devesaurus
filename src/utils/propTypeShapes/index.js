import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
