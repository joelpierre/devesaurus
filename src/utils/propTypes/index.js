import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
