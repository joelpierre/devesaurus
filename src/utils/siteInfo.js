import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
