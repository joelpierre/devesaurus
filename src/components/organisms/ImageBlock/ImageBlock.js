import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Section from '../../core/Section/Section';
import Container from '../../core/Container/Container';
import Row from '../../core/Row/Row';
import Flex from '../../core/Flex/Flex';
import Heading from '../../core/Heading/Heading';
import styles from '../CtaBanner/CtaBanner.module.scss';
import { defaultPageTheme, pageThemePropType } from '../../../utils/propTypes';

const ImageBlock = ({ module, pageTheme }) => {
  return (
    <Section
      data-test="component-image-block"
      classes={classNames([
        `theme--${module.theme ? module.theme : pageTheme}`,
        styles['image-block'],
      ])}
    >
      <Container classes="container-fluid">
        <Row classes="row">
          <Flex classes="flex text-center">
            <Heading priority="2">Image Block</Heading>
          </Flex>
        </Row>
      </Container>
    </Section>
  );
};

ImageBlock.defaultProps = {
  ...defaultPageTheme,
};

ImageBlock.propTypes = {
  module: PropTypes.instanceOf(Object).isRequired,
  ...pageThemePropType,
};

export default ImageBlock;
