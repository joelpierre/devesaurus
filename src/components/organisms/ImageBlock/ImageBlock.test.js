import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { checkProps, findByTestAttr, matchSnapshot } from '../../../utils/test';

import ImageBlock from './ImageBlock';

const defaultProps = {
  module: {
    theme: 'alpha',
  },
};

/**
 * Factory function to create a ShallowWrapper for the ImageBlock component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<ImageBlock {...setupProps} />);
};

describe('<ImageBlock/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the ImageBlock Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-image-block');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(ImageBlock, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should render with pageTheme instead of module theme', () => {
    wrapper = setup({
      module: {},
      pageTheme: 'brand',
    });
    const component = findByTestAttr(wrapper, 'component-image-block');
    matchSnapshot(wrapper);
    expect(component.prop('className')).toContain('theme--brand');
  });
});
