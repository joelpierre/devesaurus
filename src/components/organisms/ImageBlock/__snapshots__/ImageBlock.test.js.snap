// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageBlock/> should match snapshot 1`] = `
<Section
  classes="theme--alpha image-block"
  contrast={false}
  data-test="component-image-block"
  theme={null}
  type="section"
>
  <Container
    classes="container-fluid"
    fluid={false}
  >
    <Row
      classes="row"
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        auto={false}
        classes="flex text-center"
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <Heading
          classes={null}
          priority="2"
        >
          Image Block
        </Heading>
      </Flex>
    </Row>
  </Container>
</Section>
`;

exports[`<ImageBlock/> should render with pageTheme instead of module theme 1`] = `
<Section
  classes="theme--brand image-block"
  contrast={false}
  data-test="component-image-block"
  theme={null}
  type="section"
>
  <Container
    classes="container-fluid"
    fluid={false}
  >
    <Row
      classes="row"
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        auto={false}
        classes="flex text-center"
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <Heading
          classes={null}
          priority="2"
        >
          Image Block
        </Heading>
      </Flex>
    </Row>
  </Container>
</Section>
`;
